### ISSUE TITLE
Adding component to project overrides global styles

### ISSUE THREAD

**Issue by oliverodaa** [carbon-design-system/carbon-components-react#601]
## Detailed description

Describe in detail the issue you're having. Is this a feature request (new component, new icon), a bug, or a general issue?

- We use the Carbon `<Loading /> component in our project.
- Initially it did not have any styles, so we added the following line in our `styles.scss` file:
```
@import '~carbon-components/scss/components/loading/loading';
```
- Adding that line did correctly style the component, but it also changed the style of elements outside of the component.

> Is this issue related to a specific component?

I don't think it's specific to the `<Loading />` component.

> What did you expect to happen? What happened instead? What would you like to see changed?

I expected for the component to have its own independent styles.
Instead, the styles affected elements unrelated to the `<Loading />` component.

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

```
    "carbon-components": "7.23.1",
    "carbon-components-react": "3.12.4",
```

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

Nothing pressing.

## Additional information

* Screenshots or code

**Without `@import '~carbon-components/scss/components/loading/loading';`**
![screen shot 2018-02-12 at 11 57 39 am 3](https://user-images.githubusercontent.com/5886107/36111531-0ac0d3da-0fec-11e8-821e-cdf9127d7a3e.png)

**With `@import '~carbon-components/scss/components/loading/loading';`**
![screen shot 2018-02-12 at 11 56 54 am 3](https://user-images.githubusercontent.com/5886107/36111526-05186df8-0fec-11e8-8fcd-180f7d42d9e2.png)

**Comment by joshblack** [comment]
Hey @oliverodaa! Thanks so much for taking the time to make this issue, sorry about the problem you're running into!

Most likely what is happening here is that when you include the component, it's loading in the css reset from the `carbon-components` package. You can see the configuration when including carbon styles here:

https://github.com/carbon-design-system/carbon-components/blob/1a858d6ffd269b569e4303f9f1a1c1065d50e6ed/src/globals/scss/styles.scss#L5-L11

The one that's most likely relevant for you is to do add in the following **before** including the loader styles:

```scss
$css--body: false;
$css--reset: false;
```

Let me know if that helps!

**Comment by oliverodaa** [comment]
@joshblack thanks! That definitely resolves the immediate issue for our team.

May I suggest opening an issue to consider changing Carbon's approach to resetting CSS?

The reset file that has caused us problems( [_css--reset.scss](https://github.com/carbon-design-system/carbon-components/blob/f307229a33bd751ead73285d1412c81c4200f4f9/src/globals/scss/_css--reset.scss)) seems to be copy-and-pasted from http://cssreset.com/scripts/eric-meyer-reset-css/.

The original author says:
> I don’t particularly recommend that you just use this in its unaltered state in your own projects. It should be tweaked, edited, extended, and otherwise tuned to match your specific reset baseline. Fill in your preferred colors for the page, links, and so on.

It is probably a good thing to reset things like `<pre>` styles to the preferred Watson style. However it is not being reset to the preferred Watson style - it's just being reset to having no style, which at least in the case of my team has actually been harmful in the past.

**Comment by joshblack** [comment]
> Default styles like `<pre>`, `<sup>`, `<li>`, etc. should be overridden to the preferred Watson style, rather than just having all style removed.

Is this due to the ordering of how carbon is included? It seems like if the reset is included first, and then watson styles, the behavior you described should happen 😄

**Comment by oliverodaa** [comment]
> if the reset is included first, and then watson styles the behavior you described should happen

Correct. The problem is that you do not apply Watson styles.

@tw15egan previously fixed the issue for `<sup>` tags [here](https://github.com/carbon-design-system/carbon-components/issues/532).

I'm just suggesting that your team takes some time to work on the rest of the styles that you have reset but not replaced.

**Comment by joshblack** [comment]
Makes sense @oliverodaa 😄 would you mind making an issue on `carbon-components` so that we can tackle this?

**Comment by joshblack** [comment]
Thanks so much for making that issue! We really appreciate it 😄 

Hopefully we'll have that fixed soon!
